### PR TITLE
Moves over species-based unique unarmed attack effects to their limbs instead.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

While https://github.com/tgstation/tgstation/pull/70422 moved the vast majority of unarmed code to checking limbs, easy forgot one specific proc related to species. This one.

## Why It's Good For The Game

Consistency is rather important, don't you think?

## Changelog
:cl:
refactor: Moves the remaining unarmed attack based code from species to their limbs.
/:cl:

### CODE REVIEW

File: code/modules/surgery/bodyparts/species_parts/misc_bodyparts.dm [hunk 1]
```diff
@@ -469,6 +469,10 @@
 	is_dimorphic = FALSE
 	should_draw_greyscale = FALSE
 
+/obj/item/bodypart/head/golem/durathread/limb_unarmedattack_effect(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	. = ..()
```
Comment (Fikou) [inline code/modules/surgery/bodyparts/species_parts/misc_bodyparts.dm:473]: you dont need to call parent here, parent is null
also when do golems attack with their head?

Comment (necromanceranne) [reply code/modules/surgery/bodyparts/species_parts/misc_bodyparts.dm:473]: If they have monkeybrain, they bite, which uses their head as the attacking limb.

**Comment by itseasytosee** [comment]
Ah yes, consistency

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Having to copy and paste something, even if you make it a proc, on every single limb, seems like a significantly worse solution.

You need to find a better way to do this. My idea is something like:

```dm
/datum/limb_attack
    /// Limb ID to match for
    var/limb_id

/datum/limb_attack/proc/unarmed_attack_effect(mob/living/carbon/human/user, mob/living/carbon/human/target)
    return

/datum/limb_attack/golem_durathread
    limb_id = SPECIES_GOLEM_DURATHREAD

/datum/limb_attack/golem_durathread/unarmed_attack_effect(mob/living/carbon/human/user, mob/living/carbon/human/target)
    target.apply_status_effect(/datum/status_effect/strandling)
```

Then, get which limb_effect correlates to the limb, and run that proc.

**Comment by necromanceranne** [comment]
Hey I'm gonna close this one for just a little bit to free up the tracker for a while and give myself a chance at looking into what mothblocks suggested a bit more okay cool thanks sorry for the bother everyone (totally not making another pr in its place for the time being that is a smaller change)

**Pull request closed without merging**